edgeless_node/src/lib.rs

```diff
@@ -74,62 +118,68 @@ async fn fill_resources(
             .await,
         );
         provider_specifications.push(edgeless_api::node_registration::ResourceProviderSpecification {
-            provider_id: "http-ingress-1".to_string(),
+            provider_id: settings.http_ingress_provider.clone(),
             class_type: "http-ingress".to_string(),
             outputs: vec!["new_request".to_string()],
         });
     }
 
-    log::info!("Creating resource 'http-egress-1'");
-    ret.insert(
-        "http-egress-1".to_string(),
-        Box::new(
-            resources::http_egress::EgressResourceProvider::new(
-                data_plane.clone(),
-                edgeless_api::function_instance::InstanceId::new(settings.node_id.clone()),
-            )
-            .await,
-        ),
-    );
-    provider_specifications.push(edgeless_api::node_registration::ResourceProviderSpecification {
-        provider_id: "http-egress-1".to_string(),
-        class_type: "http-egress".to_string(),
-        outputs: vec![],
-    });
+    if !settings.http_egress_provider.is_empty() {
```

Comment:
Not an ideal style of defining those in a config file (new resources need new top-level options)  but nothing we have to change now.

Reply:
Totally agree, I could not think of an easy way out. Suggestions are welcome (here or as new issues).